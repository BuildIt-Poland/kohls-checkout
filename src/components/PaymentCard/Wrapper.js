import styled from 'styled-components';

const Wrapper = styled.div`
  width: ${props => (props.small ? '44px' : '132px')};
  height: ${props => (props.small ? '66px' : '172px')};
`;

export default Wrapper;
