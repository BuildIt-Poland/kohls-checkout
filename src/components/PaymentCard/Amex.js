import React from 'react';
import PropTypes from 'prop-types';

import Wrapper from './Wrapper';
import { AmexCard } from '../Icons';

function Amex({ small }) {
  return (
    <Wrapper small={small}>
      <AmexCard />
    </Wrapper>
  );
}

Amex.propTypes = {
  cardNumber: PropTypes.bool
};

export default Amex;
