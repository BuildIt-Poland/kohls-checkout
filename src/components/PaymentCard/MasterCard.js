import React from 'react';
import PropTypes from 'prop-types';

import { MasterCard as MasterCardIcon } from '../Icons';
import Wrapper from './Wrapper';

function MasterCard({ small }) {
  return (
    <Wrapper small={small}>
      <MasterCardIcon />
    </Wrapper>
  );
}

MasterCard.propTypes = {
  cardNumber: PropTypes.bool,
  className: PropTypes.string
};

export default MasterCard;
