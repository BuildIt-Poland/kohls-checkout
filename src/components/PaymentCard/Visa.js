import React from 'react';
import PropTypes from 'prop-types';

import { VisaCard } from '../Icons';
import Wrapper from './Wrapper';

function Visa({ small, className }) {
  return (
    <Wrapper className={className} small={small}>
      <VisaCard />
    </Wrapper>
  );
}

Visa.propTypes = {
  cardNumber: PropTypes.bool,
  className: PropTypes.string
};

export default Visa;
