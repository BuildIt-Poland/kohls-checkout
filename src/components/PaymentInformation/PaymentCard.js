import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import { default as DefaultPaymentCard } from '../PaymentCard';

const Wrapper = styled.div`
  grid-column-start: 1;
  grid-row-start: 1;
`;

function PaymentCard({ cardNumber }) {
  return (
    <Wrapper>
      <DefaultPaymentCard cardNumber={cardNumber} />
    </Wrapper>
  );
}

PaymentCard.propTypes = {
  cardNumber: PropTypes.string.isRequired
};

export default PaymentCard;
