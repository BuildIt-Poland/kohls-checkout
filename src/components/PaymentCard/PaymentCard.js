import React from 'react';
import PropTypes from 'prop-types';

import Visa from './Visa';
import Mastercard from './MasterCard';
import Amex from './Amex';
import { getCreditCardType } from '../../utils/creditCardType';

function PaymentCard({ cardNumber }) {
  const renderCard = () => {
    const cardType = getCreditCardType(cardNumber);

    switch (cardType) {
      case 'VISA':
        return <Visa />;
      case 'MASTER_CARD':
        return <Mastercard />;
      case 'AM_EX':
        return <Amex />;
      default:
        return null;
    }
  };

  return <>{renderCard()}</>;
}

PaymentCard.propTypes = {
  cardNumber: PropTypes.string.isRequired
};

export default PaymentCard;
