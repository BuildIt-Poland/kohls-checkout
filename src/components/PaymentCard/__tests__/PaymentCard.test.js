import React from 'react';
import { create } from 'react-test-renderer';

import PaymentCard from '../PaymentCard';

jest.mock('../../Icons', () => ({
  MasterCard: 'MasterCardIcon',
  VisaCard: 'VisaCardIcon',
  AmexCard: 'MasterCardIcon'
}));

describe('COMPONENT - PaymentCard', () => {
  it('renders PaymentCard component for incorrect card number', () => {
    const component = create(<PaymentCard cardNumber="9887786" />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders PaymentCard component for MasterCard', () => {
    const component = create(<PaymentCard cardNumber="5338876271187903" />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders PaymentCard component for American Express', () => {
    const component = create(<PaymentCard cardNumber="349061596855405" />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders PaymentCard component for Visa', () => {
    const component = create(<PaymentCard cardNumber="4485448764804164" />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
