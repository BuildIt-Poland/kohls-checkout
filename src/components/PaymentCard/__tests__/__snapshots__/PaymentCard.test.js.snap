// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - PaymentCard renders PaymentCard component for American Express 1`] = `
.c0 {
  width: 132px;
}

<div
  className="c0"
>
  <MasterCardIcon />
</div>
`;

exports[`COMPONENT - PaymentCard renders PaymentCard component for MasterCard 1`] = `
.c0 {
  width: 132px;
}

<div
  className="c0"
>
  <MasterCardIcon />
</div>
`;

exports[`COMPONENT - PaymentCard renders PaymentCard component for Visa 1`] = `
.c0 {
  width: 132px;
}

<div
  className="c0"
>
  <VisaCardIcon />
</div>
`;

exports[`COMPONENT - PaymentCard renders PaymentCard component for incorrect card number 1`] = `null`;
